retrieval/scrape.go

```diff
@@ -446,6 +461,28 @@ func (sl *scrapeLoop) stop() {
 	<-sl.done
 }
 
+func (sl *scrapeLoop) processScrapeResult(samples model.Samples, scrapeErr error, start time.Time) error {
+	// Collect samples post-relabelling and label handling in a buffer.
+	buf := &bufferAppender{buffer: make(model.Samples, 0, len(samples))}
```

Comment:
You said that already. I'm interested in the details of the setup and what was measured.
Also, just the number of allocations doesn't tell us a lot about heap fragmentation.

In different news, if 5 lines of code save 0.5% of allocation, I would go for it.